src/updater/updater.cpp

```diff
@@ -149,446 +95,568 @@ void CheckWeightsConsistency(
+                    util::Log(logWARNING)
+                        << "[weight updates] Edge weight update from "
+                        << old_fwd_durations_range[segment_offset] / 10. << "s to "
+                        << new_fwd_durations_range[segment_offset] / 10. << "s Segment: " << from
+                        << "," << to << " based on "
+                        << config.segment_speed_lookup_paths
+                               [new_fwd_datasources_range[segment_offset] - 1];
+                }
+            }
 
-                const auto segment_offset = leaf_object.fwd_segment_position;
-                const auto u_index = segment_offset;
-                const auto v_index = segment_offset + 1;
+            for (auto segment_offset = 0UL; segment_offset < new_rev_durations_range.size();
+                 ++segment_offset)
+            {
+                if (new_rev_datasources_range[segment_offset] == LUA_SOURCE)
+                    continue;
 
-                BOOST_ASSERT(u_index < nodes_range.size());
-                BOOST_ASSERT(v_index < nodes_range.size());
+                if (old_rev_durations_range[segment_offset] >=
+                    (new_rev_durations_range[segment_offset] * config.log_edge_updates_factor))
+                {
+                    auto from =
+                        internal_to_external_node_map[nodes_range[segment_offset + 1]].node_id;
+                    auto to = internal_to_external_node_map[nodes_range[segment_offset]].node_id;
+                    util::Log(logWARNING)
+                        << "[weight updates] Edge weight update from "
+                        << old_rev_durations_range[segment_offset] / 10. << "s to "
+                        << new_rev_durations_range[segment_offset] / 10. << "s Segment: " << from
+                        << "," << to << " based on "
+                        << config.segment_speed_lookup_paths
+                               [new_rev_datasources_range[segment_offset] - 1];
+                }
+            }
+        }
+    }
 
-                const extractor::QueryNode &u = internal_to_external_node_map[nodes_range[u_index]];
-                const extractor::QueryNode &v = internal_to_external_node_map[nodes_range[v_index]];
+    return updated_segments;
+}
 
-                const double segment_length = util::coordinate_calculation::greatCircleDistance(
-                    util::Coordinate{u.lon, u.lat}, util::Coordinate{v.lon, v.lat});
+void saveDatasourcesNames(const UpdaterConfig &config)
+{
+    extractor::Datasources sources;
+    DatasourceID source = 0;
+    sources.SetSourceName(source++, "lua profile");
+
+    // Only write the filename, without path or extension.
+    // This prevents information leakage, and keeps names short
+    // for rendering in the debug tiles.
+    for (auto const &name : config.segment_speed_lookup_paths)
+    {
+        sources.SetSourceName(source++, boost::filesystem::path(name).stem().string());
```

Comment:
I know you didn't change this but could you split the `source++`.
Pre / post increment magic just makes it harder to get into this code.